test/realm_test.dart

```diff
@@ -1567,6 +1566,75 @@ Future<void> main([List<String>? args]) async {
     //test the realm can be opened.
     final realm = getRealm(Configuration.disconnectedSync([Product.schema], path: path, encryptionKey: key));
   });
+
+  test('Realm.refresh', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    final result = realm.refresh();
+    expect(result, true);
+  });
+
+  test('Realm.onRefresh sync transaction', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    var called = false;
+    final transaction = realm.beginWrite();
+    realm.refreshAsync().then((_) => called = true);
+    realm.add(Person("name"));
+    transaction.commit();
+
+    await Future<void>.delayed(Duration(milliseconds: 1));
+    expect(called, true, reason: "beginWrite onRefresh failed");
+
+    called = false;
+    realm.write(() {
+      realm.refreshAsync().then((_) => called = true);
+      realm.add(Person("name"));
+    });
+
+    await Future<void>.delayed(Duration(milliseconds: 1));
+    expect(called, true, reason: "write onRefresh failed");
```

Comment:
This test looks strange to me. Why "write onRefresh failed" when called == true? Is it expected the Person not to be stored in the realm?

Reply:
This is a message if the expect fails. Not that it expect it to fail